### PR TITLE
fixes #4940 feat(nimbus): wire up end review request flow

### LINKED ISSUE #4940: Implement End UI components in front-end
In EXP-1143 we prepared the components for the End UI. From here we should be able to add them to the console.

Add this UI to the experiment Summary page, after an experiment has requested to end and the user can review the experiment. The entrypoint for this is likely inside the [EndExperiment](https://github.com/mozilla/experimenter/blob/main/app/experimenter/nimbus-ui/src/components/Summary/EndExperiment/index.tsx#L77:L77) component. Refer to the [Figma designs](https://www.figma.com/file/5SLa07XqQb0ttB1ay1gami/Nimbus-Console-UI?node-id=219024%3A3402), but take care to update the copy to not use possessive phrasing when the UI is intended for a broad audience (You can end _your_ experiment, and you can reject _the_ experiment change, but _the_ change has timed out in RS).

Throughout this flow ensure that the canReview property is being used to determine who can see what UI (unless our existing components already handle it). Currently canReview is anyone but the experiment owner, allowing the following:

* Experiment owner can see the initial live state of the experiment, where they can initiate an end request
* Experiment owner can see the rejected state/reason of an experiment
* Reviewer can see any of approve/reject UI for an experiment’s end request, the owner cannot
* Everyone can see the banner for when an end request has timed out in Remote Settings

Because the UI for this was already built in EXP-1055 the mutation for rejecting an experiment should already be in place, and completion of this issue should mean the full flow is ready for use.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-1144)
┆Issue Number: EXP-1144

### PR DESCRIPTION
**Pull request by jodyheavener**
Closes #4940 

This PR:

- Wires up the End request UI to the new `ChangeApprovalOperations` component, including adding the necessary mutations
- This change also meant gutting a lot of the `EndExperiment` component, so now it only handles the initial request kickoff
- Refactors related Storybook stories into the new ES Module format

Check out some Storybooks:

- [`Summary`](https://storage.googleapis.com/mozilla-storybooks-experimenter/commits/997b5fe1f8ddee8ba8ed0d3f8cc7005a5a2ce78b/nimbus-ui/index.html?path=/story/components-summary--draft-status)
- [`EndExperiment`](https://storage.googleapis.com/mozilla-storybooks-experimenter/commits/997b5fe1f8ddee8ba8ed0d3f8cc7005a5a2ce78b/nimbus-ui/index.html?path=/story/components-summary-endexperiment--can-request-end)

### CODE REVIEW

File: app/experimenter/kinto/tasks.py
Comment (jodyheavener) [inline app/experimenter/kinto/tasks.py:123]: Feel free to push back, but rejecting a change from the Nimbus console does _not_ prefix the message with "Rejected:", so this just makes rejections that come from RS more consistent.

**Review by lmorchard — COMMENTED** [review]
Tests pass, code looks good to me on an initial skim. Going to take a bit to run things through their paces locally and then come back for a real r+

**Comment by lmorchard** [comment]
Hmm, just pushed a preview experiment live locally & approved in Remote Settings. Got this weird state:

Capture

Going to dig around and maybe figure out why

**Comment by jodyheavener** [comment]
> Thinking we may need to tweak the queries in these latest_review_request, latest_rejection, and latest_timeout methods

Yeah I noticed this as well as stale timeout notices, and asked @jaredlockhart about them but I think we were going to do it in a followup task. I also noticed that when we started excluding DRAFT->LIVE we get stale review notices because we don't check that the last rejection changelog is the most recent changelog.

**Comment by lmorchard** [comment]
Turns out *all* the approval changelog events are stale, once I start the end experiment flow. This is a timeout from when I launched the experiment (haven't yet hit Approve and End), and the requesting user to end is actually the user who requested the launch (because I changed the user in admin)

Capture

**Comment by lmorchard** [comment]
Thinking we might need to make the queries more complex and take into account the current experiment status, not just the changelog item data.

**Comment by lmorchard** [comment]
Okay, since a backend fix occurred to me, I gave it a shot:

https://github.com/mozilla/experimenter/pull/5072

File: app/experimenter/nimbus-ui/src/components/Summary/index.tsx
Comment (jodyheavener) [reply app/experimenter/nimbus-ui/src/components/Summary/index.tsx:67]: Oooh good call. Hook incoming.

Comment (jodyheavener) [reply app/experimenter/nimbus-ui/src/components/Summary/index.tsx]: New hook: https://github.com/mozilla/experimenter/pull/5069/commits/8116c4c41a89a054b6062eccea856b2171a021e2

**Comment by jodyheavener** [comment]
I’ll hold off merging this until #5072 is in

**Comment by jaredlockhart** [comment]
We might also wanna land https://github.com/mozilla/experimenter/pull/5074 first so we can actually test the rejection flow properly

**Comment by lmorchard** [comment]
So, we landed #5072 and #5074, so I rebased this locally and gave it another run-through.

Seeing an issue as I try to end the experiment. I can initiate the review and approve the review, but it seems to jump straight to timeout before it gets to remote settings. The publish status goes Review -> Approved -> Review and never hits Waiting in-between.

Capture

**Review by lmorchard — APPROVED** [review]
Might be a few more integration wrinkles to iron out, but I don't think they're specific to this PR

**Comment by jodyheavener** [comment]
Here's a video of the whole flow from Preview to Complete (10s celery timer) that I was able to do on my machine:

https://user-images.githubusercontent.com/6392049/115565545-76105a80-a287-11eb-9e83-7cca1c41e81c.mp4

Would be good to get one other person to successfully run through it. @jaredlockhart you wanna give it a whirl?

**Comment by jaredlockhart** [comment]
I'm just going through it locally, I let it time out in RS and it it doesn't show a timeout message:

Screen Shot 2021-04-21 at 11 14 59 AM

**Comment by jaredlockhart** [comment]
And I just rejected from RS and it's not showing me a rejection message either?

Screen Shot 2021-04-21 at 11 17 32 AM

**Comment by jodyheavener** [comment]
Bleh. Okay checking it out, thank you for catching!

**Comment by jodyheavener** [comment]
Okay yeah confirmed that both of these occur for me as well. Notably a rejection from the UI _does_ correctly show up, and neither of these things occur in the approval flow.

**Comment by jaredlockhart** [comment]
We probably just need to move some statuses around.  Let me poke at it and see if I can push a small commit on top of it.

**Review by jaredlockhart — APPROVED** [review]
Yep it was just missing a changelog at completion.  All works perfectly now.  Great job @jodyheavener ! 
image

**Comment by jodyheavener** [comment]
It was so simple too 😭

**Pull request merged**
